pkg/ccl/changefeedccl/event_processing.go

```diff
@@ -626,5 +725,11 @@ func (c *parallelEventConsumer) Close() error {
 	// If an error occurred, at least one worker will return an error, so
 	// it will be returned by c.g.Wait().
 	close(c.doneCh)
-	return c.g.Wait()
+	waitErr := c.g.Wait()
+
+	// ensure goroutines blocked on inFlightWg.Wait() can exit
+	for c.inFlightCounter > 0 {
+		c.decInFlight()
+	}
```

Comment:
This felt a little hacky but I wanted to avoid requiring that we be extra careful that _every_ possible path of completing work on an event has a matching `dec`, it felt too dangerous to be relied on even if all races were fixed here.